lottie-swift/src/Public/iOS/Compatibility/CompatibleAnimationView.swift

```diff
@@ -13,27 +13,65 @@ import UIKit
 /// Use in tandem with CompatibleAnimationView when using Lottie in Objective-C
 @objc
 public final class CompatibleAnimation: NSObject {
-
+  
   @objc
   static func named(_ name: String) -> CompatibleAnimation {
     return CompatibleAnimation(name: name)
   }
-
+  @objc
+  static func filepath(_ filepath: String) -> CompatibleAnimation {
+    return CompatibleAnimation(filepath: filepath)
+  }
+  
+  @objc
+  static func json(_ json: String) -> CompatibleAnimation {
+    return CompatibleAnimation(json: json)
+  }
+  
+  @objc
+  public init(filepath: String? = nil) {
+    self.filepath = filepath
+    self.bundle = nil
+    self.name = nil
+    self.json = nil
+    super.init()
+  }
+  
+  @objc
+  public init(json: String? = nil) {
+    self.filepath = nil
+    self.bundle = nil
+    self.name = nil
+    self.json = json
+    super.init()
+  }
+  
   @objc
   public init(name: String, bundle: Bundle = Bundle.main) {
     self.name = name
     self.bundle = bundle
+    self.filepath = nil
+    self.json = nil
     super.init()
   }
 
   internal var animation: Animation? {
-    return Animation.named(name, bundle: bundle)
+    if (json != nil) {
+      let jsonData = json?.data(using: .utf8)!
```

Comment:
Be careful with `swift force unwrap`.
Would be better doing something like this:
```swift
internal var animation: Animation? {
    if let json = json {
      guard let jsonData = json.data(using: .utf8) else { return nil }
      return try? JSONDecoder().decode(Animation.self, from: jsonData)
    } else if let filepath = filepath {
      return Animation.filepath(filepath)
    } else if let name = name, let bundle = bundle {
      return Animation.named(name, bundle: bundle)
    }
    return nil
  }
```